xline-client/src/lease_gen.rs

```diff
@@ -0,0 +1,55 @@
+use std::sync::atomic::{AtomicU64, Ordering};
+
+use clippy_utilities::Cast;
+
+/// Generator of unique lease id
+/// id format:
+/// | prefix       | suffix  |
+/// | 7 bytes      | 1 byte  |
+/// | random bytes | cnt     |
+#[derive(Debug, Default)]
+pub struct LeaseIdGenerator {
+    /// prefix of id
+    prefix: u64,
+    /// suffix of id
+    suffix: AtomicU64,
+}
+
+impl LeaseIdGenerator {
+    /// New `IdGenerator`
+    ///
+    /// # Panics
+    ///
+    /// panic if failed to generate random bytes
+    #[inline]
+    #[must_use]
+    pub fn new() -> Self {
+        let mut buf = [0u8; 8];
+        getrandom::getrandom(&mut buf).unwrap_or_else(|err| {
+            panic!("Failed to generate random bytes for lease id generator: {err}");
+        });
+        let prefix = u64::from_be_bytes(buf).overflowing_shl(1).0;
+        let suffix = AtomicU64::new(0);
+        Self { prefix, suffix }
+    }
+
+    /// Generate next id
+    #[inline]
+    pub fn next(&self) -> i64 {
```

Comment:
Why do we return a `i64` instead of `u64`? Can a lease id be negative?

Reply:
The lease ID defined in protobuf api is i64. However only zero and positive value are used, I guess the negative id is reserved.